app/drivers/uart.h

```diff
@@ -4,6 +4,11 @@
 #include "errors.h"
 
 #define EUART_HAL_INIT					(EUART_BASE + 0)
+#define EUART_TX_SEMPH					(EUART_BASE + 1)
+#define EUART_TX						(EUART_BASE + 2)
+#define EUART_RX						(EUART_BASE + 3)
 
 UART_HandleTypeDef *uart_get_handle(void);
+err_t uart_tx(UART_HandleTypeDef *p_handle, const uint8_t *p_buf, uint32_t size, uint32_t timeout_ms);
+err_t uart_rx(UART_HandleTypeDef *p_handle, uint8_t *p_buf, uint32_t size, uint32_t timeout_ms);
```

Comment:
maybe we should use (freertos-)ticks instead for the timeout parameters.

Reply:
UT the Hal uses ms so we would convert one way and then back to pass to the Hal?